### PR TITLE
AVM8 callstack frames update

### PR DESCRIPTION
**Pull request by nwatson22**

### CODE REVIEW

File: lib/include/kframework/avm/teal/teal-stack.md [hunk 1]
```diff
@@ -55,6 +55,25 @@ The elements of the stack are values of sort `TValue`, i.e. either `TUInt64` or
...
+- `S { N }` accesses element $N$ (counting from the bottom of the stack) of $S$.
+```k
+  syntax TValue ::= TStack "{" Int "}" [function]
+  // -----------------------------------------------------------
+  rule (X : XS):TStack { N } => X
+    requires N ==Int #sizeTStack(XS)
+  rule XS              { N } => #drop((#sizeTStack(XS) -Int N) -Int 1, XS) [ 0 ]
+    requires N  <Int #sizeTStack(XS)
+
+//  rule XS:TStack {N} => #sizeTStack(#drop(#sizeTStack(XS) -Int N, XS))
```
Comment (geo2a) [inline lib/include/kframework/avm/teal/teal-stack.md:67]: Why is this line commented out? Should we just drop it?

File: lib/include/kframework/avm/teal/teal-stack.md [hunk 2]
```diff
@@ -55,6 +55,25 @@ The elements of the stack are values of sort `TValue`, i.e. either `TUInt64` or
                                                    andBool N  <Int #sizeTStack(XS)
 ```
 
+- `S { N }` accesses element $N$ (counting from the bottom of the stack) of $S$.
+```k
+  syntax TValue ::= TStack "{" Int "}" [function]
+  // -----------------------------------------------------------
```
Comment (geo2a) [inline lib/include/kframework/avm/teal/teal-stack.md:61]:
```suggestion
  syntax TValue ::= TStack "{" Int "}" [function]
  // --------------------------------------------
```

File: lib/include/kframework/avm/teal/teal-stack.md [hunk 3]
```diff
@@ -55,6 +55,25 @@ The elements of the stack are values of sort `TValue`, i.e. either `TUInt64` or
...
+  rule XS              { N } => #drop((#sizeTStack(XS) -Int N) -Int 1, XS) [ 0 ]
+    requires N  <Int #sizeTStack(XS)
+
+//  rule XS:TStack {N} => #sizeTStack(#drop(#sizeTStack(XS) -Int N, XS))
+```
+
+- `S { N <- M }` set element $N$ (counting from the bottom of the stack) of $S$ to $M$.
+```k
+  syntax TStack ::= TStack "{" Int "<-" TValue "}" [function]
+  // -----------------------------------------------------------
```
Comment (geo2a) [inline lib/include/kframework/avm/teal/teal-stack.md:73]:
```suggestion
  syntax TStack ::= TStack "{" Int "<-" TValue "}" [function]
  // --------------------------------------------------------
```

**Review by geo2a — CHANGES_REQUESTED** [review]

**Comment by geo2a** [comment]
Looks good to me overall, but please address the nitpicks.

File: lib/include/kframework/avm/teal/teal-stack.md (hunk 1 above)
Comment (nwatson22) [reply lib/include/kframework/avm/teal/teal-stack.md]: I removed it.

**Comment by nwatson22** [comment]
@geo2a There is some sort of issue in the CI with the python linting, although the python code was not modified in this PR. I was not able to replicate it locally. I wonder if the CI has a different pyk version than I do, but I'm not sure how to check that.

File: tests/teal-sources/frame-access1.teal [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#pragma version 7
```
Comment (nwatson22) [inline tests/teal-sources/frame-access1.teal:1]:
```suggestion
#pragma version 8
```

File: tests/teal-sources/subroutine.teal [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#pragma version 7
```
Comment (nwatson22) [inline tests/teal-sources/subroutine.teal:1]:
```suggestion
#pragma version 8
```

File: tests/teal-sources/frame-access2.teal [hunk 6]
```diff
@@ -0,0 +1,34 @@
+#pragma version 7
```
Comment (nwatson22) [inline tests/teal-sources/frame-access2.teal:1]:
```suggestion
#pragma version 8
```

File: tests/teal-sources/frame-access3.teal [hunk 7]
```diff
@@ -0,0 +1,22 @@
+#pragma version 7
```
Comment (nwatson22) [inline tests/teal-sources/frame-access3.teal:1]:
```suggestion
#pragma version 8
```

**Comment by geo2a** [comment]
Sorry about that @nwatson22, the linting is fixed now. There's one spec that does not pass  though, please have a look.

**Review by geo2a — APPROVED** [review]

**Pull request merged**